lib/Clift/CliftTypes.cpp

```diff
@@ -85,6 +88,23 @@ static void mlir::printCliftDebugName(mlir::AsmPrinter &Printer,
   }
 }
 
+static mlir::ParseResult mlir::parseCliftComment(mlir::AsmParser &Parser,
+                                                 std::string &Comment) {
+  if (Parser.parseOptionalKeyword("commented-with").succeeded()) {
+    if (Parser.parseString(&Comment).failed())
+      return mlir::failure();
+  }
+  return mlir::success();
+}
+
+static void mlir::printCliftComment(mlir::AsmPrinter &Printer,
```

Comment:
Commenting here, but actually relevant for all users of this (and of the `parse` equivalent).

You always print comments after the debug name, but I'd like them printed at the end.
Otherwise, if you're reading some IR with lots of comments it's going to become pretty unreadable pretty soon.
If you print comments at the end, things are much more manageable.